src/Facets/EcoFacet.sol

```diff
@@ -164,151 +190,58 @@ contract EcoFacet is ILiFi, ReentrancyGuard, SwapperV2, Validatable, LiFiData {
-                value: 0
-            });
-        } else {
-            routeCalls[0] = IEcoPortal.Call({
-                target: _ecoData.receiverAddress,
-                data: "",
-                value: _bridgeData.minAmount
-            });
-        }
-
-        return routeCalls;
-    }
-
-    function _buildRouteTokens(
-        ILiFi.BridgeData memory _bridgeData,
-        EcoData calldata _ecoData
-    ) private pure returns (IEcoPortal.TokenAmount[] memory) {
-        // Route tokens are the tokens needed on the destination chain
-        // If receiving native, no tokens needed
-        if (LibAsset.isNativeAsset(_ecoData.receivingAssetId)) {
-            return new IEcoPortal.TokenAmount[](0);
-        }
-
-        // If receiving ERC20, we need those tokens on destination
-        IEcoPortal.TokenAmount[]
-            memory routeTokens = new IEcoPortal.TokenAmount[](1);
-        routeTokens[0] = IEcoPortal.TokenAmount({
-            token: _ecoData.receivingAssetId,
-            amount: _bridgeData.minAmount
-        });
-
-        return routeTokens;
-    }
-
-    function _buildIntent(
-        ILiFi.BridgeData memory _bridgeData,
-        EcoData calldata _ecoData,
-        IEcoPortal.Call[] memory routeCalls,
-        IEcoPortal.TokenAmount[] memory routeTokens
-    ) private view returns (IEcoPortal.Intent memory) {
-        bool isNative = LibAsset.isNativeAsset(_bridgeData.sendingAssetId);
-
-        // Build reward tokens array for ERC20
-        IEcoPortal.TokenAmount[] memory rewardTokens;
-        if (!isNative) {
-            rewardTokens = new IEcoPortal.TokenAmount[](1);
-            rewardTokens[0] = IEcoPortal.TokenAmount({
-                token: _bridgeData.sendingAssetId,
-                amount: _bridgeData.minAmount + _ecoData.solverReward
-            });
-        } else {
-            rewardTokens = new IEcoPortal.TokenAmount[](0);
-        }
-
-        return
-            IEcoPortal.Intent({
-                route: IEcoPortal.Route({
-                    salt: _ecoData.salt,
-                    source: block.chainid,
-                    destination: _bridgeData.destinationChainId,
-                    inbox: _ecoData.destinationInbox,
-                    tokens: routeTokens,
-                    calls: routeCalls
-                }),
-                reward: IEcoPortal.Reward({
-                    creator: msg.sender, // Address that can cancel/retrieve refund
-                    prover: _ecoData.prover,
-                    deadline: _ecoData.rewardDeadline,
-                    nativeValue: isNative
-                        ? _ecoData.solverReward + _bridgeData.minAmount
-                        : 0, // No native amount for ERC20
-                    tokens: rewardTokens // Include token reward for ERC20
-                })
-            });
-    }
 
     function _publishIntent(
         ILiFi.BridgeData memory _bridgeData,
         EcoData calldata _ecoData,
         IEcoPortal.Intent memory intent
```

Comment:
```suggestion
        Reward memory reward
```
You don't event use intent.destination or intent.route, so just pass the reward in to make it clearer.

Reply:
Actually, don't even need to pass reward in. Just call _buildReward here with the `_bridgeData` and `_ecoData`.